src/components/dashboard/WhoContent.native.js

```diff
@@ -0,0 +1,195 @@
+
+  const getAllContacts = () => {
+    Contacts.getAll((err, contacts) => {
+      if (err === 'denied') {
+        console.warn('permissions denied')
+      } else {
+        const sortContacts = orderBy(contacts, ['givenName'])
+        setContacts(sortContacts)
+        setInitalList(sortContacts)
+      }
+    })
+  }
+
+  const getUserFeed = async () => {
+    const userFeed = await userStorage.getFeedPage(5, true)
+    const recent = userFeed.filter(({ type }) => type === 'send' || type === 'receive')
+    setRecentFeedItems(recent)
+  }
+
+  const showPermissionsAndroid = () => {
+    PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.READ_CONTACTS, {
+      title: 'Contacts',
+      message: 'We need access to view your contacts, so you can easily send G$ to them.',
+      buttonPositive: 'Approve',
+    }).then(() => {
+      getAllContacts()
+    })
+  }
+
+  useEffect(() => {
+    getUserFeed()
+  }, [])
+
+  useEffect(() => {
+    if (Contacts) {
+      if (isAndroid) {
+        showPermissionsAndroid()
+      } else {
+        getAllContacts()
+      }
+    }
+  }, [])
+
+  const handleSearch = useCallback(query => {
+    const queryIsNumber = parseInt(query)
+    if (state) {
+      setValue(query)
+    }
+    if (query && !query.includes('+') && !query.includes('*')) {
+      if (typeof queryIsNumber === 'number' && !isNaN(queryIsNumber)) {
+        return setContacts(
+          initialList.filter(({ phoneNumbers }) => get(phoneNumbers, '[0].number', '').indexOf(query) >= 0)
+        )
+      }
+      if (typeof query === 'string') {
+        return setContacts(
+          initialList.filter(({ givenName, familyName }) => {
+            const fullName = `${givenName} ${familyName}`.toLocaleLowerCase()
+            return fullName.indexOf(query.toLocaleLowerCase()) >= 0
+          })
+        )
+      }
+    } else {
+      getAllContacts()
+    }
+  })
+
+  useEffect(() => {
+    if (contacts.length === 0) {
+      showNext(true)
+    } else {
+      showNext(false)
+    }
+  }, [contacts])
+
+  useEffect(() => {
+    findRecentlyUsed()
+  }, [contacts, recentFeedItems])
+
+  const findRecentlyUsed = () => {
+    let matches = []
+    if (recentFeedItems.length > 0 && contacts.length > 0) {
+      recentFeedItems.forEach(feedItem => {
+        matches.push(
+          contacts.filter(
+            contact => contact.phoneNumbers[0] && feedItem.data.phoneNumber === contact.phoneNumbers[0].number
+          )
+        )
+      })
+    }
+
+    setRecentlyUsedList(matches.flat())
+  }
+
+  return (
+    <>
+      <Section.Stack justifyContent="space-around" style={styles.container}>
+        <Section.Title fontWeight="medium">{text}</Section.Title>
+        <InputText
+          error={error}
+          onChangeText={handleSearch}
+          placeholder="Search contact name / phone"
+          style={styles.input}
+          value={value}
+          enablesReturnKeyAutomatically
+          onSubmitEditing={next}
```

Comment:
not sure i understand this.
next should be called when a contact is set, this input is to update the contact list.
i think next is not supposed to be passed from parent anymore

Reply:
Yes, you have the reason, this should be only for web